components/disk/component.go

```diff
@@ -340,6 +467,63 @@ func (c *component) Check() components.CheckResult {
 
 	log.Logger.Debugw(cr.reason, "extPartitions", len(cr.ExtPartitions), "nfsPartitions", len(cr.NFSPartitions), "blockDevices", len(cr.BlockDevices))
 
+	// Check for disk failure events from kmsg (only if event store is available)
+	if c.eventBucket != nil && c.rebootEventStore != nil {
+		// Query recent events to check for disk failures
+		recentEvents, err := c.eventBucket.Get(c.ctx, cr.ts.Add(-c.lookbackPeriod))
+		if err != nil {
+			log.Logger.Warnw("failed to get recent events", "error", err)
+		} else {
+			// Filter recent events to only include the hardware-related disk failure events
+			var diskFailureEvents eventstore.Events
+			var failureReasons []string
+			for _, ev := range recentEvents {
+				switch ev.Name {
+				case eventRAIDArrayFailure:
+					diskFailureEvents = append(diskFailureEvents, ev)
+					cr.health = apiv1.HealthStateTypeUnhealthy
+					failureReasons = append(failureReasons, "RAID array failure detected")
+				case eventFilesystemReadOnly:
+					diskFailureEvents = append(diskFailureEvents, ev)
+					cr.health = apiv1.HealthStateTypeUnhealthy
+					failureReasons = append(failureReasons, "Filesystem remounted read-only")
+				case eventNVMePathFailure:
+					diskFailureEvents = append(diskFailureEvents, ev)
+					cr.health = apiv1.HealthStateTypeUnhealthy
+					failureReasons = append(failureReasons, "NVMe path failure detected")
+				}
+			}
+
+			// Append failure reasons to existing reason if any failures were detected
+			if len(failureReasons) > 0 {
+				newReason := strings.Join(failureReasons, ", ")
+				if cr.reason != "" {
+					cr.reason += "; " + newReason
+				} else {
+					cr.reason = newReason
+				}
+			}
+
+			// If we found disk failures, evaluate suggested actions
+			if len(diskFailureEvents) > 0 {
+				// The kmsg syncer already inserted the event, so we just need to handle the health state
+				// and evaluate suggested actions based on historical data
+
+				// Look up past events to derive the suggested actions
+				rebootEvents, err := c.rebootEventStore.GetRebootEvents(c.ctx, cr.ts.Add(-c.lookbackPeriod))
+				if err != nil {
+					cr.err = err
+					cr.health = apiv1.HealthStateTypeUnhealthy
+					log.Logger.Warnw("failed to get reboot events", "error", cr.err)
+					return cr
+				}
+
+				// Evaluate the suggested actions (along with the reboot history)
+				evaluateSuggestedActions(cr, rebootEvents, diskFailureEvents)
```

Comment:
diskFailureEvents contains three kinds of event, and it's not reasonable to evaluate suggested action with it..